docs/reference-manual/native-image/guides/optimize-file-size.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: ni-docs
+toc_group: how-to-guides
+link_title: Optimize a Native Executable for File Size
+permalink: /reference-manual/native-image/guides/optimize-for-file-size/
+---
+
+# Optimize a Native Executable for File Size
+
+You can optimize your native executable by taking advantage of different optimization levels. 
+This guide will demonstrate how to create small native executables for a given application, using the optimization for size, `-Os`.
+
+> `-Os` enables `-O2` optimizations except those that can increase code or executable size significantly. Typically, it creates the smallest possible executables at the cost of reduced performance. Learn more in [Optimization Levels](../OptimizationsAndPerformance.md#optimization-levels).
+
+### Prerequisite 
+
+Make sure you have installed Oracle GraalVM for JDK 23 or later.
+The easiest way to get started is with [SDKMAN!](https://sdkman.io/jdks#graal){:target="_blank"}.
+For other installation options, visit the [Downloads](https://www.graalvm.org/downloads/){:target="_blank"} section.
+
+For the demo, run a "fortune teller" application that simulates the traditional `fortune` Unix program (for more information, see [fortune](https://en.wikipedia.org/wiki/Fortune_(Unix)){:target="_blank"}).
+
+1. Clone the GraalVM Demos repository:
+    ```bash
+    git clone https://github.com/graalvm/graalvm-demos.git
+    ```
+    
+2. Change directory to _fortune-demo/fortune-maven_:
+    ```bash
+    cd fortune-demo/fortune-maven
+    ```
+
+## Build a Native Executable with Default Configuration
+
+1. Create a native executable using the [Maven plugin for Native Image](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html){:target="_blank"}:
+    ```bash
+    mvn -Pnative package
+    ```
+    The command compiles the project, creates a JAR file with all dependencies, and then generates a native executable, `fortune`, in the _target_ directory.
+
+2. (Optional) Run the application:
+    ```bash
+    ./target/fortune
+    ```
+    The application will return a random saying. 
+
+3. Check the file size which should be around 13M:
+    ```bash
+    du -sh target/fortune*
```

Comment:
```suggestion
    du -h target/fortune
```

Isn't the use of `-s` (and the glob) misleading here? The actual size of the native image is the size of the binary only.